chain/types/actor_event.go

```diff
@@ -37,25 +27,30 @@ type ActorEventFilter struct {
 	// If the value is an empty slice, the filter will match on the key only, accepting any value.
 	Fields map[string][]ActorEventBlock `json:"fields,omitempty"`
 
-	// Interpreted as an epoch (in hex) or one of "latest" for last mined block, "earliest" for first,
-	// Optional, default: "latest".
-	FromEpoch string `json:"fromEpoch,omitempty"`
+	// The height of the earliest tipset to include in the query. If empty, the query starts at the
+	// last finalized tipset.
+	// NOTE: In a future upgrade, this will be strict when set and will result in an error if a filter
+	// cannot be fulfilled by the depth of history available in the node. Currently, the node will
+	// nott return an error, but will return starting from the epoch it has data for.
+	FromHeight *abi.ChainEpoch `json:"fromHeight,omitempty"`
 
-	// Interpreted as an epoch (in hex) or one of "latest" for last mined block, "earliest" for first,
-	// Optional, default: "latest".
-	ToEpoch string `json:"toEpoch,omitempty"`
+	// The height of the latest tipset to include in the query. If empty, the query ends at the
+	// latest tipset.
+	ToHeight *abi.ChainEpoch `json:"toHeight,omitempty"`
 
 	// Restricts events returned to those emitted from messages contained in this tipset.
-	// If `TipSetCid` is present in the filter criteria, then neither `FromEpoch` nor `ToEpoch` are allowed.
-	TipSetCid *cid.Cid `json:"tipsetCid,omitempty"`
+	// If `TipSetKey` is legt empty in the filter criteria, then neither `FromHeight` nor `ToHeight` are allowed.
+	TipSetKey *TipSetKey `json:"tipsetKey,omitempty"`
```

Comment:
Pointer here may not be idiomatic; I just wanted to avoid an unnecessary `[]` in the json form, mainly. I think maybe if `[]` is omitted on the input side that it may still resolve to the zero-value here, I'm not sure.